about/about-kubernetes-networking.md

```diff
@@ -0,0 +1,79 @@
+---
+title: About Kubernetes Networking
+description: Learn about Kubernetes networking!
+---
+
+Kubernetes defines a network model that helps provide simplicity and consistency across a range of networking environments and network implementations. The Kubernetes network model provides the foundation for understanding how containers, pods, and services within Kubernetes communicate with each other. This guide explains the key concepts and how they fit together. 
+
+In this guide you will learn:
+- The fundamental network behaviors the Kubernetes network model defines.
+- How Kubernetes works with a variety of different network implementations.
+- What Kubernetes Services are.
+- How DNS works within Kubernetes.
+- What "NAT outgoing" is and when you would want to use it.
+- What "dual stack" is.
+
+### The Kubernetes network model
+
+The Kubernetes network model specifies:
+- Every pod gets its own IP address
+- Containers within a pod share the pod IP address and can communicate freely with each other
+- Pods can communicate with all other pods in the cluster using pod IP addresses (without [NAT]({{site.baseurl}}/about/about-networking#nat))
+- Isolation (restricting what each pod can communicate with) is defined using network policies
+
+As a result, pods can be treated much like VMs or hosts (they all have unique IP addresses), and the containers within pods very much like processes running within a VM or host (they run in the same network namespace and share an IP address). This model makes it easier for applications to be migrated from VMs and hosts to pods managed by Kubernetes. In addition, because isolation is defined using network policies rather than the structure of the network, the network remains simple to understand. This style of network is sometimes referred to as a "flat network".
+
+Note that, although very rarely needed, Kubernetes does also support the ability to map host ports through to pods, or to run pods directly within the host network namespace sharing the host's IP address.
+
+### Kubernetes network implementations
+
+Kubernetes built in network support, kubenet, can provide some basic network connectivity. However, it is more common to use third party network implementations which plug into Kubernetes using the CNI (Container Network Interface) API.
+
+There are two basic kinds of CNI plugins:
+- network plugins, which are responsible for connecting pod to the network
+- IPAM (IP Address Management) plugins, which are responsible for allocating pod IP addresses.
```

Comment:
I'd be tempted to omit the bit about IPAM plugins?  I'm not sure it's that relevant to people, and there are other types of plugins too.
- Meta-plugins like Multus
- Not-sure-of-category-name-but-stuff-modifying plugins like portmap, bandwidth and tuning.

Reply:
I also don't feel like this really helps clarify the interaction with cloud provider plugins, where the key thing is that Calico CNI can operate in both policy-only and networking-and-policy modes.

Reply:
I reworded it slightly to emphasize there are many kinds of plugins, but I do want to mention IPAM because it is a feature that many users do care about, and it applies to all clusters.